### PR TITLE
fix depth check to prevent max call stack exceeded issue

### PR DESCRIPTION
**Pull request by omarrida**

### CODE REVIEW

File: test-definitions/fern/apis/circular-references/definition/ast.yml [hunk 1]
```diff
@@ -14,3 +9,14 @@ types:
...
+  FieldName: string
+  FieldValue:
+    union:
+      primitive_value: PrimitiveValue
+      object_value: ObjectValue
+      container_value: ContainerValue
+  ObjectFieldValue:
+    properties:
+      name: FieldName
+      value: FieldValue
```
Comment (dsinghvi) [inline test-definitions/fern/apis/circular-references/definition/ast.yml:21]: is it worth capturing the edge case this was testing?

**Review by armandobelardo — APPROVED** [review]
Approving given seed passes locally

**Review by dsinghvi — APPROVED** [review]
